api-get-object.go

```diff
@@ -289,6 +291,21 @@ func (o *Object) doGetRequest(request getRequest) (getResponse, error) {
 		return response, response.Error
 	}
 
+	// Check if ETag has changed since last request.
+	switch {
+	// If response doesn't carry a valid ETag, ignore.
+	case response.objectInfo.ETag == "":
+
+	// Set Object's etag on first request.
+	case o.etag == "":
+		o.etag = response.objectInfo.ETag
+
+	// Check if etag from response matches Object's etag.
+	case o.etag != response.objectInfo.ETag:
+		return response, errors.New("object has been modified since last read")
```

Comment:
> You can use the ErrorResponse thingy here and return BadDigest instead.

@harshavardhana, `The Content-Md5 you specified did not match what we received.` can be misleading to someone receiving this error for `Read` or `ReadAt` on `minio.Object`. Further, the request headers sent during a `GetObject` were correct, at that point in time.Thoughts?